nada_dsl/nada_types/collections.py

```diff
@@ -501,24 +598,12 @@ def new(cls, *args) -> "Array[T]":
             ),
         )
 
-    @classmethod
-    def generic_type(cls, contained_type: T, size: int) -> ArrayType:
-        """Return the generic type of the Array."""
-        return ArrayType(contained_type=contained_type, size=size)
-
     @classmethod
     def init_as_template_type(cls, contained_type) -> "Array[T]":
         """Construct an empty template array with the given child type."""
         return Array(child=None, contained_type=contained_type, size=None)
 
 
-@dataclass
-class VectorType(Collection):
-    """The generic type for Vectors."""
-
-    contained_type: AllTypesType
-
-
 @dataclass
 class Vector(Generic[T], Collection):
```

Comment:
Maybe, we should remove Vector. It's not fully supported and it's not in the roadmap. Thoughts @navasvarela @lumasepa

Reply:
That's a good point. I agree we should remove it since it is not used anywhere. We can always add it in the future if we need it.

Reply:
Sounds good to me. I'll create another PR for that though.

Reply:
https://github.com/NillionNetwork/nada-dsl/pull/58